wrappers/rust/icicle-core/src/curve.rs

```diff
@@ -147,13 +149,18 @@ where
     type ArkEquivalent = ArkAffine<C::ArkSWConfig>;
 
     fn to_ark(&self) -> Self::ArkEquivalent {
-        let proj_x = self
-            .x
-            .to_ark();
-        let proj_y = self
-            .y
-            .to_ark();
-        Self::ArkEquivalent::new_unchecked(proj_x, proj_y)
+        if *self == Self::zero() {
+            Self::ArkEquivalent::zero()
```

Comment:
is it more efficient than the else case?

Reply:
The else case just doesn't cover zero. `new_unchecked` assumes that inputs represent a valid non-zero point.